### PR TITLE
Unify behavior between asgi and wsgi 

### LINKED ISSUE #1679: Unify behavior between asgi and wsgi
Currently the implementation of the response class for the wsgi and the asgi differ when handling the `data` property. Namely the asgi version does not override data with the media value, while the wsgi version overrides the data with the media serialization data.

In the code on master there are currently these todos
https://github.com/falconry/falcon/blob/c3e91875f1030c68d17efa27cf8df3d18f337761/falcon/response.py#L192-L195
https://github.com/falconry/falcon/blob/c3e91875f1030c68d17efa27cf8df3d18f337761/falcon/response.py#L192-L195

Should the wsgi response class behave like the asgi, and use the method `render_body` to get the body in the response?

### PR DESCRIPTION
**Pull request by CaselIT**
# Summary of Changes

- avoid duplication of `Request.prefix`
- wgsi `Request` now has a `get_media` method
- wsgi `Response` now has a `render_body` methos and behaves like asgi `Response` regarding the data and media properties

I'm not sure where and how the documentation should be updated. Do you have suggestions?

# Related Issues

Fixes #1679

# Pull Request Checklist

This is just a reminder about the most common mistakes.  Please make sure that you tick all *appropriate* boxes.  But please read our [contribution guide](../CONTRIBUTING.md) at least once; it will save you a few review cycles!

If an item doesn't apply to your pull request, **check it anyway** to make it apparent that there's nothing to do.

- [x] Applied changes to both WSGI and ASGI code paths and interfaces (where applicable).
- [x] Added **tests** for changed code.
- [x] Prefixed code comments with GitHub nick and an appropriate prefix.
- [x] Coding style is consistent with the rest of the framework.
- [ ] Updated **documentation** for changed code.
    - [x] Added docstrings for any new classes, functions, or modules.
    - [x] Updated docstrings for any modifications to existing code.
    - [x] Updated both WSGI and ASGI docs (where applicable).
    - [x] Added references to new classes, functions, or modules to the relevant RST file under `docs/`.
    - [x] Updated all relevant supporting documentation files under `docs/`.
    - [x] A copyright notice is included at the top of any new modules (using your own name or the name of your organization).
    - [x] Changed/added classes/methods/functions have appropriate `versionadded`, `versionchanged`, or `deprecated` [directives](http://www.sphinx-doc.org/en/stable/usage/restructuredtext/directives.html?highlight=versionadded#directive-versionadded).
- [x] Changes (and possible deprecations) have [towncrier](https://towncrier.readthedocs.io/en/actual-freaking-docs/index.html) news fragments under `docs/_newsfragments/`, with the file name format `{issue_number}.{fragment_type}.rst`. (Run `towncrier --draft` to ensure it renders correctly.)

If you have *any* questions to *any* of the points above, just **submit and ask**! This checklist is here to *help* you, not to deter you from contributing!

*PR template inspired by the attrs project.*

### CODE REVIEW

File: falcon/constants.py [hunk 1]
```diff
@@ -106,3 +106,7 @@
     'referer',
     'user-agent',
 ])
+
+# NOTE(kgriffs): Special singleton to be used internally whenever using
+#   None would be ambiguous.
+_UNSET = object()
```
Comment (CaselIT) [inline falcon/constants.py]: Moved here otherwise it cannot be imported by `.request.Response`

**Comment by kgriffs** [comment]
> I'm not sure where and how the documentation should be updated. Do you have suggestions?

No problem, I can just add a patch to include the doc updates. Everything else looks great, thanks!

**Comment by kgriffs** [comment]
Docs updated. This should be ready for final review. @vytas7

**Comment by CaselIT** [comment]
The change in docs looks good!

**Comment by CaselIT** [comment]
nice!

**Pull request merged**